shapely/ops.py

```diff
@@ -210,26 +183,24 @@ def voronoi_diagram(geom, envelope=None, tolerance=0.0, edges=False):
     [1] https://en.wikipedia.org/wiki/Voronoi_diagram
     [2] https://geos.osgeo.org/doxygen/geos__c_8h_source.html  (line 730)
     """
-    func = lgeos.methods['voronoi_diagram']
-    envelope = envelope._geom if envelope else None
     try:
-        result = geom_factory(func(geom._geom, envelope, tolerance, int(edges)))
-    except ValueError:
-        errstr = "Could not create Voronoi Diagram with the specified inputs."
+        result = pygeos.voronoi_polygons(
+            geom, tolerance=tolerance, extend_to=envelope, only_edges=edges
+        )
+    except pygeos.GEOSException as err:
+        errstr = "Could not create Voronoi Diagram with the specified inputs "
+        errstr += "({}).".format(str(err))
         if tolerance:
             errstr += " Try running again with default tolerance value."
-        raise ValueError(errstr)
+        raise pygeos.GEOSException(errstr) from err
```

Comment:
I'm OK with exposing GEOSException, but I thinks @sgillies has preferred raising a Shapely error instead?

Reply:
For this specific case we have easy control over it (since we are catching the original GEOSException anyway to add some context to the error message), but the general issue of raising GEOSException to the user or not is a broader discussion, and some discussion is in https://github.com/Toblerity/Shapely/issues/991

Reply:
I very much prefer to raise an error that derives from `ShapelyError` (https://github.com/Toblerity/Shapely/blob/master/shapely/errors.py#L4). How about a `VoronoiDiagramError` ?

Reply:
Note that `GEOSException` is "our own exception class", just like `ShapelyError` is (once pygeos is incorporated into shapely).

I would prefer if we can then revive the discussion in https://github.com/Toblerity/Shapely/issues/991, which exactly deals with this topic. 
I could certainly create a custom VoronoiDiagramError for this case (to get this PR merged), but it would then be adding a special case, since most other operations raise GEOSExceptions in the shapely-2.0 branch. So personally I would rather first get to conclusion for the general discussion.

Reply:
Ping here. Is it OK to keep this discussion for #991, and move forward with this PR? 

I will keep the current ValueError instead of changing it to GEOSException, then this PR is just the status quo on that aspect.

Reply:
Thanks @jorisvandenbossche !